python/xir/program.py

```diff
@@ -201,19 +203,31 @@ class Declaration:
 
     Args:
         name (str): name of the declaration
+        declaration_type (str): The type of declaration. Can be either "gate", "operator", "output"
+            or "function".
+        params (Sequence[str]): parameters used by the declared object
+        wires (Sequence[Wire]): wires that the declared object is applied to
     """
 
     def __init__(
-        self, name: str, params: Sequence[str], wires: Sequence[Wire], declaration_type: str
+        self,
+        name: str,
+        declaration_type: str,
+        params: Optional[Sequence[str]] = None,
+        wires: Optional[Sequence[Wire]] = None,
```

Comment:
This is nice; however, the number of lines that had to be changed as a result highlights why calling Python functions using the keyword argument syntax can be preferable to using positional arguments.